### DISCUSSION TITLE
Expose Projection with a flag incompleteSiblingData !!

### DISCUSSION THREAD

**Discussion by geminigeek** [payloadcms/payload#1612] [Feature Requests & Ideas]
hi,

i suggest an option to expose projection for mongoose queries, as payload hooks make use of sibling data we can set another flag  "incompleteSiblingData" whenever a projection key is set in the find query, this way hooks logic will know that we do not have all the sibling data.

my use case or where it will be beneficial ? 

i have a collection with 60 fields per doc most of the fields are only for internal/admin use and contain very large text , which almost every time i query with payload local api, comes from disk from mongodb server, while the data sent to frontend is all in indexed field, if i run queries with projection natively using req.payload.collections the query runs faster, as less data is pulled from disk  and transferred between db and Api server , also no need to post process much before saving to cache layer!

found similar issue https://github.com/payloadcms/payload/discussions/1042

**Comment by DanRibbens** [comment]
I could see how this would be useful. My initial thought is it could be a fairly simple change, though there may be some other ramifications.

Would it be better to pass the `projection` object to the hooks instead of `incompleteSiblingData`?
This allows the same check for truthyness. Then it would also let you know whether a field is actually undefined or was omitted from the query.

Is this something you would like to make a PR for? I'd like to get other feedback first.

We also have a PR to add a `select` option to relationships, somewhat related. #1288
I'm thinking about the names here being different and having similar functionality.

**Reply by geminigeek** [reply]
Hi Dan,

i can't  make PR as i am new to JavaScript ,  still learning bits and pieces ,  let other users make comments regarding this, but payload is an excellent tool as it exposes mongoose so most of the functionalities that are not natively available in payload can be done via mongoose model , for few queries i am using mongoose model directly to achieve result as i suggested above , my queries are now blazing fast with projection
